src/plugins/data/common/expressions/interpreter.ts

```diff
@@ -17,13 +17,18 @@
  * under the License.
  */
 
-// All types must be universal and be castable on the client or on the server
 import { get } from 'lodash';
-import { getType } from '@kbn/interpreter/common';
 
-// TODO: Currently all casting functions must be syncronous.
+function getType(node: any) {
+  if (node == null) return 'null';
+  if (typeof node === 'object') {
+    if (!node.type) throw new Error('Objects must have a type property');
+    return node.type;
+  }
+  return typeof node;
```

Comment:
nit: I noticed `getType` was rewritten twice: once in `common/serialize` and once in `common/type`.

If this is just being copied over from `packages/kbn-interpreter`, I wonder if we should leave everything as-is for now, and then in a separate PR update all imports of `getType` function to pull from the plugin instead of from the package (canvas uses this in several places too). That way we don't have 3 different copies of `getType` floating around in the repo.

Reply:
In the NP plugin the idea is not to depend on `@kbn/interpreter` at all. That's is why currently it is inlined, but in the future I was discussing with @ppisljar that we don't really need `Registry` class from `@kbn/interpreter` at all, that it could be replaced simply by `new Map()`, so the idea is that `Registry` will go away, I don't know if `getType` will go away, but if it does not, it will live in this `./common/expressions` folder instead of `@kbn/interpreter`.

With regards to Canvas, they can use `@kbn/interpreter` if they wish, but there is probably no reason to do so. Because all 3 interpreter related registries are now in OSS interpreter/expressions plugins, and the 9 registries they use `Registry` from `@kbn/intepreter` for I believe have nothing to do with interpreter.